### ISSUE TITLE
Storing special chars crashes

### ISSUE THREAD

**Issue by Casperhr** [vapor/mysql#42]
Fx storing the string
```
"KMD IRONMAN Denmark added 399 new photos to the album: IRONKIDS Copenhagen 2016 — with Julie Damstedt and Simon Smith."
```

Will crash at Bind.swift
```
  public convenience init(_ string: String) {
        let bytes = Array(string.utf8)
        let buffer = UnsafeMutablePointer<Char>.allocate(capacity: bytes.count)
        for (i, byte) in bytes.enumerated() {
            buffer[i] = Char(byte) //HERE
        }

        self.init(type: MYSQL_TYPE_STRING, buffer: buffer, bufferLength: bytes.count)
    }
```

**Comment by nickskull** [comment]
Just to clarify, it's probably the unicode long dash causing it "—", code \u2014.

**Comment by tannernelson** [comment]
Can you include the stack trace?

**Comment by Casperhr** [comment]
Jep it's code the byte is >128 so it crashes

```
#0	0x0000000100707e29 in Bind.init(String) -> Bind at /Users/casper_r/vapor/ironman/Packages/MySQL-0.4.0/Sources/MySQL/Bind.swift:71
#1	0x0000000100708063 in Bind.__allocating_init(String) -> Bind ()
#2	0x000000010070923f in Node.bind.getter at /Users/casper_r/vapor/ironman/Packages/MySQL-0.4.0/Sources/MySQL/Bind.swift:198
#3	0x000000010070c378 in Binds.(init([NodeRepresentable]) throws -> Binds).(closure #1) at /Users/casper_r/vapor/ironman/Packages/MySQL-0.4.0/Sources/MySQL/Binds.swift:22
#4	0x000000010070c50a in thunk ()
#5	0x000000010070c619 in partial apply for thunk ()
#6	0x0000000100779076 in Collection.map<A> ((A.Iterator.Element) throws -> A1) throws -> [A1] ()
#7	0x000000010070c113 in Binds.init([NodeRepresentable]) throws -> Binds at /Users/casper_r/vapor/ironman/Packages/MySQL-0.4.0/Sources/MySQL/Binds.swift:22
#8	0x000000010070c22e in Binds.__allocating_init([NodeRepresentable]) throws -> Binds ()
#9	0x0000000100711275 in Database.execute(String, [NodeRepresentable], Connection?) throws -> [[String : Node]] at /Users/casper_r/vapor/ironman/Packages/MySQL-0.4.0/Sources/MySQL/Database.swift:128
#10	0x0000000100740308 in MySQLDriver.mysql(String, [Node], Connection?) throws -> Node at /Users/casper_r/vapor/ironman/Packages/FluentMySQL-0.4.1/Sources/FluentMySQL/MySQLDriver.swift:110
#11	0x000000010073e53a in MySQLDriver.query<A where ...> (Query<A>) throws -> Node at /Users/casper_r/vapor/ironman/Packages/FluentMySQL-0.4.1/Sources/FluentMySQL/MySQLDriver.swift:67
#12	0x0000000100740cb9 in protocol witness for Driver.query<A where ...> (Query<A1>) throws -> Node in conformance MySQLDriver ()
#13	0x000000010017e103 in Query.raw() throws -> Node at /Users/casper_r/vapor/ironman/Packages/Fluent-0.9.0/Sources/Fluent/Query/Query.swift:79
#14	0x000000010017fe45 in QueryRepresentable.create(Node?) throws -> Node at /Users/casper_r/vapor/ironman/Packages/Fluent-0.9.0/Sources/Fluent/Query/Query.swift:151
#15	0x0000000100180bd3 in QueryRepresentable.save(inout A.T) throws -> () at /Users/casper_r/vapor/ironman/Packages/Fluent-0.9.0/Sources/Fluent/Query/Query.swift:170
#16	0x000000010016a3cc in Entity.save() throws -> () at /Users/casper_r/vapor/ironman/Packages/Fluent-0.9.0/Sources/Fluent/Entity/Entity.swift:46
```

**Comment by collinhundley** [comment]
This is the same as #25, which should have been fixed with v0.4. @Casperhr can you verify in your code that `Char` is a typealias for `UInt8` and not `Int8`?

**Comment by nickskull** [comment]
@collinhundley I've just verified and it shows up as typealias for `Int8`. Although, this project is still using Vapor 0.16.

**Comment by collinhundley** [comment]
That's strange because your stack trace shows v0.4, but somehow you've got old code in there. Either way, updating should do the trick :)

**Comment by tannernelson** [comment]
This should be fixed in 0.5 and later. Feel free to re-open if it's still happening.
